server/src/main/java/org/jboss/as/server/moduleservice/ServiceModuleLoader.java

```diff
@@ -218,33 +218,45 @@ public static void installModuleResolvedService(ServiceTarget serviceTarget, Mod
      * until all the transitive dependencies that it depends upon have have their module spec services
      * come up.
      *
-     * @param identifier The module identifier
+     * @param name The module name
      * @return The service name of the ModuleSpec service
      */
-    public static ServiceName moduleResolvedServiceName(ModuleIdentifier identifier) {
-        if (!isDynamicModule(identifier)) {
-            throw ServerLogger.ROOT_LOGGER.missingModulePrefix(identifier, MODULE_PREFIX);
+    public static ServiceName moduleResolvedServiceName(String name) {
```

Comment:
This falls into the same category as what was done with `ServiceName moduleServiceName(ModuleIdentifier identifier)` and should be deprecated as well, although better to get other feedback since I don't know exactly what was the use case where this can be called from other parties than WildFly